src/sandbox.js

```diff
@@ -1,71 +1,152 @@
+import 'regenerator-runtime/runtime';
 import React, { useEffect } from 'react';
 import ReactDOM from 'react-dom';
 import Scrollable from './components/Scrollable';
 import setupHighlighter from '../devtools/src/content-script/highlighter';
 import cssPath from './lib/cssPath';
 import { getQueryAdvise } from './lib';
+import debounce from 'lodash.debounce';
+import parser from './parser';
 
-function postMessage(event, payload) {
+const state = {
+  queriedNodes: [],
+  markup: '',
+  query: '',
+  rootNode: null,
+  highlighter: null,
+};
+
+function postMessage(action) {
   top.postMessage(
     {
       source: 'testing-playground-sandbox',
-      event,
-      payload,
+      ...action,
     },
     top.location.origin,
   );
 }
 
-let rootNode;
-let highlighter;
+function runQuery(rootNode, query) {
+  const result = parser.parse({ rootNode, query });
+  state.queriedNodes = result.elements.map((elem) => elem.target);
+  state.highlighter.highlight({ nodes: state.queriedNodes });
+}
+
+function setInnerHTML(node, html) {
+  const doc = node.ownerDocument;
+  node.innerHTML = html;
+
+  for (let prevScript of node.querySelectorAll('script')) {
+    const newScript = doc.createElement('script');
```

Comment:
In this way, aren't you getting (and replacing with js functions) all the `script` elements, not only the `javascript` ones?

Reply:
Thanks :+1: . Fixed